src/GafferImage/Resize.cpp

```diff
@@ -0,0 +1,250 @@
+}
+
+Gaffer::StringPlug *Resize::filterPlug()
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 2 );
+}
+
+const Gaffer::StringPlug *Resize::filterPlug() const
+{
+	return getChild<StringPlug>( g_firstPlugIndex + 2 );
+}
+
+Gaffer::AtomicBox2fPlug *Resize::dataWindowPlug()
+{
+	return getChild<AtomicBox2fPlug>( g_firstPlugIndex + 3 );
+}
+
+const Gaffer::AtomicBox2fPlug *Resize::dataWindowPlug() const
+{
+	return getChild<AtomicBox2fPlug>( g_firstPlugIndex + 3 );
+}
+
+Resample *Resize::resample()
+{
+	return getChild<Resample>( g_firstPlugIndex + 4 );
+}
+
+const Resample *Resize::resample() const
+{
+	return getChild<Resample>( g_firstPlugIndex + 4 );
+}
+
+void Resize::affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const
+{
+	ImageProcessor::affects( input, outputs );
+
+	if( input == formatPlug() )
+	{
+		outputs.push_back( outPlug()->formatPlug() );
+		outputs.push_back( dataWindowPlug() );
+	}
+	else if(
+		input == inPlug()->formatPlug() ||
+		input == fitPlug() ||
+		input == inPlug()->dataWindowPlug()
+	)
+	{
+		outputs.push_back( dataWindowPlug() );
+	}
+	else if(
+		input == inPlug()->channelDataPlug() ||
+		input == filterPlug()
+	)
+	{
+		outputs.push_back( outPlug()->channelDataPlug() );
+	}
+}
+
+void Resize::hash( const ValuePlug *output, const Context *context, IECore::MurmurHash &h ) const
+{
+	ImageProcessor::hash( output, context, h );
+
+	if( output == dataWindowPlug() )
+	{
+		formatPlug()->hash( h );
+		fitPlug()->hash( h );
+		inPlug()->formatPlug()->hash( h );
+		inPlug()->dataWindowPlug()->hash( h );
+	}
+}
+
+void Resize::compute( ValuePlug *output, const Context *context ) const
+{
+	if( output == dataWindowPlug() )
+	{
+		const Format inFormat = inPlug()->formatPlug()->getValue();
+		const Format outFormat = formatPlug()->getValue();
+
+		const V2f inSize( inFormat.width(), inFormat.height() );
+		const V2f outSize( outFormat.width(), outFormat.height() );
+		const V2f formatScale = outSize / inSize;
+
+		V2f dataWindowScale( 1 );
+		switch( (Fit)fitPlug()->getValue() )
+		{
+			case Horizontal :
+				dataWindowScale = V2f( formatScale.x );
+				break;
+			case Vertical :
+				dataWindowScale = V2f( formatScale.y );
+				break;
+			case Best :
+				dataWindowScale = V2f( std::min( formatScale.x, formatScale.y ) );
+				break;
+			case Fill :
+				dataWindowScale = V2f( std::max( formatScale.x, formatScale.y ) );
+				break;
+			case Distort :
+			default :
+				dataWindowScale = formatScale;
+				break;
+		}
+
+		const V2f dataWindowOffset = ( outSize - ( inSize * dataWindowScale ) ) / 2.0f;
+		const Box2i inDataWindow = inPlug()->dataWindowPlug()->getValue();
+		const Box2f outDataWindow(
+			V2f( inDataWindow.min ) * dataWindowScale + dataWindowOffset,
+			/// \todo Adjust for #1438
+			V2f( inDataWindow.max + V2i( 1 ) ) * dataWindowScale + dataWindowOffset - V2f( 1.0f )
+		);
+
+		static_cast<AtomicBox2fPlug *>( output )->setValue( outDataWindow );
```

Comment:
Somewhere in here is some funky float rounding...

If I take an inDataWindow of (0,0)-(199,149), and do a Horizontal Resize to 1920x1080, the outDataWindow here comes out as (-6.10325e-05,-180)-(1919,1259)

All is right except for that minuscule offset in min.x, meaning that the actual data window ends up going from -1 in x. Can this round to the nearest, say, 0.01, or something small enough to make no difference?